api/v1alpha1/envoyproxy_types.go

```diff
@@ -137,13 +137,25 @@ type ShutdownConfig struct {
 // EnvoyProxyKubernetesProvider defines configuration for the Kubernetes resource
 // provider.
 type EnvoyProxyKubernetesProvider struct {
+	// Mode is the type of the Kubernetes resource provider. If unspecified, it defaults to "Deployment".
```

Comment:
do we need an explicit mode field or can we use CEL validation to make sure 
* if `Deployment` is set, then `DaemonSet` cannot be set

Reply:
If in future someone add tag `+kubebuilder:default:="****"` to field in `KubernetesDeploymentSpec` or `KubernetesDaemonSetSpec`, then field `envoyDaemonSet` or `envoyDaemonSet`will not be empty. How to implement this case?

+ the implicit behavior is a bit confusing

Reply:
will that be rejected by validating webhook?

Reply:
I'm not sure I understand what you're referring to. Does Envoy Gateway implement an Admission (Validation) webhook?

Reply:
EG use [CEL](https://kubernetes.io/docs/reference/using-api/cel/)

Reply:
@arkodg Can you help here? Which kubebuilder tag need you for this logic?

Reply:
1. use a ptr type called `mode` to decide b/w Deployment and DaemonSet, defaulting to Deployment
2. use CEL to make sure when `mode` is set to `DaemonSet` , user cannot populate the deployment field e.g. https://github.com/envoyproxy/gateway/blob/0f87fcbad1aaa2913af3f768da87ea72a695a14d/api/v1alpha1/ext_auth_types.go#L13

Reply:
> do we need an explicit mode field or can we use CEL validation to make sure

May be we don't need `mode`? Only `EnvoyDeployment` and `EnvoyDaemonSet` field with validation. If not set - use Deployment (with default preset)

Reply:
> > do we need an explicit mode field or can we use CEL validation to make sure
> 
> May be we don't need `mode`? Only `EnvoyDeployment` and `EnvoyDaemonSet` fields with validation. If not set - use Deployment (with default preset)

I prefer this approach